src/plugins/data/public/search/errors/painless_error.tsx

```diff
@@ -67,21 +66,10 @@ export class PainlessError extends KbnError {
   }
 }
 
-function getFailedShards(err: EsError) {
-  const failedShards =
-    err.body?.attributes?.error?.failed_shards ||
-    err.body?.attributes?.error?.caused_by?.failed_shards;
-  return failedShards ? failedShards[0] : undefined;
-}
-
-function getRootCause(err: EsError) {
-  return getFailedShards(err)?.reason;
-}
-
-export function isPainlessError(err: Error | EsError) {
+export function isPainlessError(err: Error | IEsError) {
   if (!isEsError(err)) return false;
 
-  const rootCause = getRootCause(err as EsError);
+  const rootCause = getRootCause(err as IEsError);
```

Comment:
Is the cast here necessary (since you've checked `isEsError` above, which is implemented as a type check)?